verifiers/internal/gha/slsaprovenance/v1.0/byob.go

```diff
@@ -0,0 +1,268 @@
+package v1
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	slsa1 "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v1"
+	serrors "github.com/slsa-framework/slsa-verifier/v2/errors"
+
+	"github.com/slsa-framework/slsa-verifier/v2/verifiers/internal/gha/slsaprovenance/common"
+)
+
+var (
+	// byobBuildType is the base build type for BYOB delegated builders.
+	byobBuildType = "https://github.com/slsa-framework/slsa-github-generator/delegator-generic@v0"
+
+	// containerBasedBuildType is the build type for the container-based builder and is based on BYOB.
+	containerBasedBuildType = "https://slsa.dev/container-based-build/v0.1?draft"
+
+	// genericGHABuildType is used by some tests.
+	// TODO: Update tests to use a real buildType.
+	genericGHABuildType = "https://github.com/Attestations/GitHubActionsWorkflow@v1"
+)
+
+// BYOBProvenanceV1 is SLSA v1.0 provenance for the slsa-github-generator BYOB build type.
+type BYOBProvenanceV1 struct {
+	prov *intotoAttestation
+}
+
+// Predicate implements ProvenanceV02.Predicate.
+func (p *BYOBProvenanceV1) Predicate() slsa1.ProvenancePredicate {
+	return p.prov.Predicate
+}
+
+// BuilderID implements Provenance.BuilderID.
+func (p *BYOBProvenanceV1) BuilderID() (string, error) {
+	return p.prov.Predicate.RunDetails.Builder.ID, nil
+}
+
+// SourceURI implements Provenance.SourceURI.
+func (p *BYOBProvenanceV1) SourceURI() (string, error) {
+	// Use resolvedDependencies.
```

Comment:
Right, maybe I can just update this to include the refactor and hold off on the changes for the sha1 support since the uri & branch/tag will likely need to come from that rather than from `resolvedDependencies`.